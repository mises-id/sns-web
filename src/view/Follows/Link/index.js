/*
 * @Author: lmk
 * @Date: 2021-07-15 01:09:29
 * @LastEditTime: 2022-02-14 10:20:29
 * @LastEditors: lmk
 * @Description:
 */

import React from "react";
import link from "@/images/link.png";
import Image from "@/components/Image/index";
import "./index.scss";
// import { sdkLocationHref } from '@/utils/postMessage';
/**
 * @description: userHeader and follow Btn
 * @param {*} theme
 * @return {*} element
 */
const Link = ({ theme = "primary", item = {}, type }) => {
  const bgClass = {
    primary: "m-bg-f8f8f8",
    white: "m-bg-fff",
  }[theme];
  const linkTo = (e) => {
    e.stopPropagation();
    window.open(item.link);
  };
  return (
    item && (
      <div
        className={`m-flex m-row-between forwardBox ${bgClass}`}
        onClick={linkTo}
      >
        <div className="m-flex">
          <Image
            shape="square"
            size={!type ? "sm" : 35}
            alt="image"
            source={item.attachment_url}
          />
          <div className="m-margin-left7">
            <span className={`title ${!type ? "m-font12" : "m-font15"}`}>
              {item.title}
            </span>
            <p
              className={`m-colors-999 timeAndType ${
                !type ? "m-font10" : "m-font13"
              }`}
            >
              {item.host}
            </p>
          </div>
        </div>
        <img src={link} className="iconStyle m-margin-right5" alt="link"></img>
      </div>
    )
  );
};
export default Link;
