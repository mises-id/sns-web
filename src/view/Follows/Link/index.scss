.iconStyle{
  width: 15px;
  height: 15px;
}
.forwardBox{
  border-radius: 3px;
  padding: 6px;
  margin-bottom: 5px;
}