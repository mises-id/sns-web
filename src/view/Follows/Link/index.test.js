

import React from 'react';
import {shallow } from 'enzyme';
import Link from './';
describe('Link', () => {
  it('should render correctly', () => {
    const component = shallow(<Link />)
    expect(component).toMatchSnapshot();
  });
});