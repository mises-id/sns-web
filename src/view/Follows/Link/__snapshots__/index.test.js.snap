// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
<div
  className="m-flex m-row-between forwardBox m-bg-f8f8f8"
  onClick={[Function]}
>
  <div
    className="m-flex"
  >
    <Image
      alt="image"
      shape="square"
      size="sm"
    />
    <div
      className="m-margin-left8"
    >
      <span
        className="m-font12 title"
      />
      <p
        className="timeAndType m-colors-999"
      />
    </div>
  </div>
  <img
    alt="link"
    className="iconStyle m-margin-right5"
    src="link.png"
  />
</div>
`;
